### PR TITLE
docs: add math structure diagram

### PR DESCRIPTION
**Pull request by Insun35**
Add math structure diagram.

### CODE REVIEW

**Comment by chokobole** [comment]
Could you more specify `Multivariate` since we have MLE and multivariate polynomial? Also it would be awesome you can express something like `UnivariateDensePolynomial` as an example of implementaitons!

File: docs/how_to_contribute/math/math_structure.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+This is an abstract diagram of the **math** directory structure.
+
+![image](/docs/how_to_contribute/math/math_structure.png)
+
+Math structure is hierarchically organized from **Semigroup** to **Group**, **Ring**, **Field**, sequentially inheriting from one another.
+
+- [**Semigroup**](https://mathworld.wolfram.com/Semigroup.html): A Semigroup <S, ○> is a mathematical object defined for a set 'S' and a binary operator '○' in which the operation is associative.
+- [**Group**](https://mathworld.wolfram.com/Group.html): A Group 'G' is a set of elements combined with a binary operation (called the group operation) that together satisfy the four fundamental properties: closure, associative, identity, and inverses. The **Elliptic Curve** inherits the Group class.
+- [**Ring**](https://mathworld.wolfram.com/Ring.html): A Ring is a set 'S' with operations + and \*. The **Polynomial** inherits the Ring class. The Ring class satisfies the followings:
```
Comment (ashjeong) [inline docs/how_to_contribute/math/math_structure.md:11]:
```suggestion
- [**Ring**](https://mathworld.wolfram.com/Ring.html): A Ring is a set 'S' with operations + and \*. **Polynomials** inherit the Ring class. The Ring class satisfies the following:

File: docs/how_to_contribute/math/math_structure.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Math Structure
+
+This is an abstract diagram of the **math** directory structure.
+
+![image](/docs/how_to_contribute/math/math_structure.png)
+
+Math structure is hierarchically organized from **Semigroup** to **Group**, **Ring**, **Field**, sequentially inheriting from one another.
+
+- [**Semigroup**](https://mathworld.wolfram.com/Semigroup.html): A Semigroup <S, ○> is a mathematical object defined for a set 'S' and a binary operator '○' in which the operation is associative.
+- [**Group**](https://mathworld.wolfram.com/Group.html): A Group 'G' is a set of elements combined with a binary operation (called the group operation) that together satisfy the four fundamental properties: closure, associative, identity, and inverses. The **Elliptic Curve** inherits the Group class.
```
Comment (ashjeong) [inline docs/how_to_contribute/math/math_structure.md:10]:
```suggestion
- [**Group**](https://mathworld.wolfram.com/Group.html): A Group 'G' is a set of elements combined with a binary operation (called the group operation) that together satisfy the four fundamental properties: closure, associative, identity, and inverses. **Elliptic Curves** inherit the Group class.

File: docs/how_to_contribute/math/math_structure.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+- [**Ring**](https://mathworld.wolfram.com/Ring.html): A Ring is a set 'S' with operations + and \*. The **Polynomial** inherits the Ring class. The Ring class satisfies the followings:
+  1. Additive associativity: (a + b) + c = a + (b + c)
+  2. Additive commutativity: a + b = b + a
+  3. Additive identity: a + 0 = 0 + a = a
+  4. Additive inverse: a + (-a) = (-a) + a = 0
+  5. Distributivity: a \* (b + c) = (a \* b) + (a \* c)
+  6. Multiplicative associativity: (a \* b) \* c = a \* (b \* c)
+- [**Field**](https://mathworld.wolfram.com/Field.html): A Field is any set of elements that satisfies the field axioms for both addition and multiplication and is commutative division algebra. Simply put, a field is a ring in which multiplicative commutativity exists, and every non-zero element has a multiplicative inverse. The **Finite Field** inherits the Field class.
```
Comment (ashjeong) [inline docs/how_to_contribute/math/math_structure.md:18]:
```suggestion
- [**Field**](https://mathworld.wolfram.com/Field.html): A Field is any set of elements that satisfies the field axioms for both addition and multiplication and is commutative division algebra. Simply put, a field is a ring in which multiplicative commutativity exists, and every non-zero element has a multiplicative inverse. **Finite Fields** inherit the Field class.

File: docs/how_to_contribute/math/math_structure.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Math Structure
+
+This is an abstract diagram of the **math** directory structure.
+
+![image](/docs/how_to_contribute/math/math_structure.png)
+
+Math structure is hierarchically organized from **Semigroup** to **Group**, **Ring**, **Field**, sequentially inheriting from one another.
```
Comment (ashjeong) [inline docs/how_to_contribute/math/math_structure.md:7]:
```suggestion
The math structure is hierarchically organized from **Semigroup** to **Group**, **Ring**, **Field**, sequentially inheriting from one another.

**Review by ashjeong — COMMENTED** [review]
Cool diagram!

File: docs/how_to_contribute/math/math_structure.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+Multivariate polynomials can be represented in multivariate sparse coefficient form and multilinear dense evaluation form. Since the multivariate dense coefficient form and multilinear spare evaluation form rarely exist, neither are implemented.
+
+## Elliptic Curves
+
+[**Elliptic Curves**](https://mathworld.wolfram.com/EllipticCurve.html) inherit the Group class. Elliptic curves come in various forms, but all currently implemented elliptic curves are in the Short Weierstrass form. Elliptic curves' points can be represented in four formats: Affine, Projective, Jacobian, and PointXYZZ.
+
+The currently implemented elliptic curves are BLS12-381, BN254, Secp256k1, and Pasta. Similar to finite fields, we generate code with Bazel's generator to add a wider variety of curves. For instructions on how to add a new elliptic curve, refer to this [doc](/docs/how_to_contribute/math/how_to_add_elliptic_curves.md).
+
+Since the result of AffinePoint + JacobianPoint is a JacobianPoint, the `ResultTy` of `AdditiveSemigroupTraits` for the elliptic curves is a JacobianPoint.
```
Comment (ashjeong) [inline docs/how_to_contribute/math/math_structure.md:35]: I'm not too sure what this is trying to tell us-it seems to come out of nowhere

Comment (Insun35) [reply docs/how_to_contribute/math/math_structure.md]: We decided to add only the diagram in this PR and update this doc in another PR.

**Review by chokobole — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**